cpp/tests/binaryop/util/operation.h

```diff
@@ -0,0 +1,161 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cmath>
+#include <type_traits>
+#include <cstdint>
+
+namespace cudf {
+namespace library {
+namespace operation {
+
+    template <typename TypeOut, typename TypeLhs, typename TypeRhs>
+    struct Add {
+        TypeOut operator()(TypeLhs lhs, TypeRhs rhs) {
+            using TypeCommon = typename std::common_type<TypeLhs, TypeRhs>::type;
+            return (TypeOut)((TypeCommon)lhs + (TypeCommon)rhs);
+        }
+    };
+
+    template <typename TypeOut, typename TypeLhs, typename TypeRhs>
+    struct Sub {
+        TypeOut operator()(TypeLhs lhs, TypeRhs rhs) {
+            using TypeCommon = typename std::common_type<TypeLhs, TypeRhs>::type;
+            return (TypeOut)((TypeCommon)lhs - (TypeCommon)rhs);
+        }
+    };
+
+    template <typename TypeOut, typename TypeLhs, typename TypeRhs>
+    struct Mul {
+        TypeOut operator()(TypeLhs lhs, TypeRhs rhs) {
+            using TypeCommon = typename std::common_type<TypeLhs, TypeRhs>::type;
+            return (TypeOut)((TypeCommon)lhs * (TypeCommon)rhs);
+        }
+    };
+
+    template <typename TypeOut, typename TypeLhs, typename TypeRhs>
+    struct Div {
+        TypeOut operator()(TypeLhs lhs, TypeRhs rhs) {
+            using TypeCommon = typename std::common_type<TypeLhs, TypeRhs>::type;
+            return (TypeOut)((TypeCommon)lhs / (TypeCommon)rhs);
+        }
+    };
+
+    template <typename TypeOut, typename TypeLhs, typename TypeRhs>
+    struct TrueDiv {
+        TypeOut operator()(TypeLhs lhs, TypeRhs rhs) {
+            return (TypeOut)((double)lhs / (double)rhs);
+        }
+    };
+
+    template <typename TypeOut, typename TypeLhs, typename TypeRhs>
+    struct FloorDiv {
+        TypeOut operator()(TypeLhs lhs, TypeRhs rhs) {
+            return (TypeOut)floor((double)lhs / (double)rhs);
+        }
+    };
+
+    template <typename TypeOut,
+              typename TypeLhs,
+              typename TypeRhs,
+              typename Common = typename std::common_type<TypeLhs, TypeRhs>::type>
+    struct Mod;
+
+    template <typename TypeOut, typename TypeLhs, typename TypeRhs>
+    struct Mod<TypeOut, TypeLhs, TypeRhs, int64_t> {
+        TypeOut operator()(TypeLhs x, TypeRhs y) {
+            return (TypeOut)((int64_t)x % (int64_t)y);
+        }
+    };
+
+    template <typename TypeOut, typename TypeLhs, typename TypeRhs>
+    struct Mod<TypeOut, TypeLhs, TypeRhs, float> {
+        TypeOut operator()(TypeLhs x, TypeRhs y) {
+            return (TypeOut)fmod((float)x, (float)y);
+        }
+    };
+
+    template <typename TypeOut, typename TypeLhs, typename TypeRhs>
+    struct Mod<TypeOut, TypeLhs, TypeRhs, double> {
+        TypeOut operator()(TypeLhs x, TypeRhs y) {
```

Comment:
This is a method of a stateless object. No reason why it shouldn't be `const`. 

Also, please consider using explicit C++-style casts `static_cast<>, const_cast<>, dynamic_cast<>`, etc.